### PR TITLE
Delete and recreate all affected directories during an artifact cache refresh

### PR DESCRIPTION
**Pull request by jason-simmons**
This will ensure cleanup of any files that existed in previous versions of
the artifacts but have since been removed

### CODE REVIEW

File: packages/flutter_tools/lib/src/cache.dart [hunk 1]
```diff
@@ -315,39 +317,36 @@ class FlutterEngine {
...
 
-    bool allDirty = engineVersion != cache.getStampFor(kName);
-
     Directory pkgDir = cache.getCacheDir('pkg');
     for (String pkgName in _getPackageDirs()) {
       Directory dir = new Directory(path.join(pkgDir.path, pkgName));
-      if (!dir.existsSync() || allDirty) {
-        await _downloadItem('Downloading package $pkgName...', url + pkgName + '.zip', pkgDir);
-      }
+      if (dir.existsSync())
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/cache.dart]: Is `allDirty` used anymore?

**Comment by devoncarew** [comment]
lgtm

File: packages/flutter_tools/lib/src/cache.dart (hunk 1 above)
Comment (jason-simmons) [inline packages/flutter_tools/lib/src/cache.dart]: allDirty was removed

Comment (devoncarew) [inline packages/flutter_tools/lib/src/cache.dart]: Ah, ok, I see that above.

**Pull request merged**